token/rust/src/client.rs

```diff
@@ -0,0 +1,235 @@
+use async_trait::async_trait;
+use solana_client::rpc_client::RpcClient;
+use solana_program_test::{tokio::sync::Mutex, BanksClient, ProgramTestContext};
+use solana_sdk::{
+    account::Account, hash::Hash, pubkey::Pubkey, signature::Signature, transaction::Transaction,
+};
+use std::{fmt, future::Future, pin::Pin, sync::Arc};
+
+type BoxFuture<'a, T> = Pin<Box<dyn Future<Output = T> + Send + 'a>>;
+
+/// Basic trait for sending transactions to validator.
+pub trait SendTransaction {
```

Comment:
Hmm, it sort of seems like almost everything in this module is generic enough to be upstreamed somewhere; maybe a new spl-client crate or even solana-client? I'm not sure what is the best fit yet.
Since I don't want to hold this up on that decision or a potential new solana-client release, how about we rename everything here to remove Token references, and then it will be easy to drop-in replace after I upstream?

Reply:
Also, bump on this

Reply:
Not sure that I understand what should be renamed :disappointed:

Reply:
> Not sure that I understand what should be renamed

Everything in this client.rs that has the string `Token` in it, since none of it is specific to Token. I'll make a couple suggestions below as examples.
I'm not totally set on the prefix `Program`, so if you think of a better generic let me know.